### DISCUSSION TITLE
Missing newline character at beginning of values

### DISCUSSION THREAD

**Discussion by tom42** [VerifyTests/Verify#1152] [General]
This class

```
public sealed class TestClass
{
    public string Property1 { get; set; } = "Text without newline character(s)";
    public string Property2 { get; set; } = "Text with\nnewline character(s)";
    public string Property3 { get; set; } = "\nText with\nnewline character(s)";
}
```

when run through through this test using Verify.Xunit 23.1.0 using default verifier settings:

```
[Fact]
public Task Test()
{
    return Verifier.Verify(new TestClass());
}
```

yields the following _received_ file:

```
{
  Property1: Text without newline character(s),
  Property2:
Text with
newline character(s),
  Property3:
Text with
newline character(s)
}
```

The newline character at the beginning of strings matters to me, so I'd expect the received file to look like this:

```
{
  Property1: Text without newline character(s),
  Property2:
Text with
newline character(s),
  Property3:

Text with
newline character(s)
}
```

Am I missing something?

**Comment by SimonCropp** [comment]
verify tries to make the output more readable. the newline logic is part of that. if you prefer to disable that behaviour try strict json https://github.com/VerifyTests/Verify/blob/main/docs/serializer-settings.md#usestrictjson

**Comment by tom42** [comment]
Thank you for the quick reply. Strict JSON might be an option, will look into that.

No offence meant, but I find the current logic odd nevertheless. If I add to the class above another property with two newline characters at the beginning:
```
public string Property4 { get; set; } = "\n\nText with\nnewline character(s)";
```

Then I get the following received file:
```
{
  Property1: Text without newline character(s),
  Property2:
Text with
newline character(s),
  Property3:
Text with
newline character(s),
  Property4:

Text with
newline character(s)
}
```
It seems like only the first newline character is removed. This might be rather subjective, but if it's supposed to be more readable, then I'd expect Property3 and Property4 to be the same. As it is I find it confusing.

**Comment by SimonCropp** [comment]
happy to consider a pull request that adds an API to opt in to the behavior you desire
